### PR TITLE
Add tests for s3 download url

### PR DESCRIPTION
**Pull request by bogdan-zaharia**
## Link to JIRA ticket

https://hackney.atlassian.net/jira/software/projects/DES/boards/45?selectedIssue=DES-57

### *What changes have we introduced*

- Added tests to cover the s3 download url feature (gateway tests, use case tests, e2e tests)

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: DocumentsApi/V1/UseCase/DownloadDocumentUseCase.cs [hunk 1]
```diff
@@ -26,7 +26,18 @@ public string Execute(string documentId)
...
-            return _s3Gateway.GeneratePreSignedDownloadUrl(document);
+            var result = "";
+
+            try
+            {
+                result = _s3Gateway.GeneratePreSignedDownloadUrl(document);
+            }
+            catch (ArgumentException e)
+            {
+                Console.WriteLine("Error when retrieving the presigned URL: '{0}' ", e.Message);
```
Comment (neilmendum) [inline DocumentsApi/V1/UseCase/DownloadDocumentUseCase.cs:37]: I think you need to `throw e` so the `ClaimsController` can catch it

File: DocumentsApi.Tests/V1/E2ETests/ClaimsTest.cs [hunk 2]
```diff
@@ -104,5 +104,40 @@ public async Task FindingAnInvalidDocumentReturns404()
...
+                          "}";
+
+            var jsonString = new StringContent(body, Encoding.UTF8, "application/json");
+            var response = await Client.PostAsync(uri, jsonString).ConfigureAwait(true);
+
+            response.StatusCode.Should().Be(201);
+        }
+
+        [Test]
+        public async Task Returns404WhenCannotFindDocument()
```
Comment (neilmendum) [inline DocumentsApi.Tests/V1/E2ETests/ClaimsTest.cs:128]: I think we'll also want a test for `Returns404WhenArgumentExceptionThrown`

File: DocumentsApi/V1/UseCase/DownloadDocumentUseCase.cs (hunk 1 above)
Comment (bogdan-zaharia) [reply DocumentsApi/V1/UseCase/DownloadDocumentUseCase.cs:40]: 👍

**Review by neilmendum — APPROVED** [review]
LGTM!

**Pull request merged**